tools/shared.py

```diff
@@ -65,6 +65,7 @@
 diagnostics.add_warning('version-check')
 diagnostics.add_warning('export-main')
 diagnostics.add_warning('unused-command-line-argument', shared=True)
+diagnostics.add_warning('pthreads-mem-growth', enabled=True, part_of_all=False)
```

Comment:
`enabled` defaults to true so I don't think you need that.  And I think you probably what part_of_all to just default to True too.

Reply:
I think this is ready to land aside from this comment?

Reply:
I have addressed the comments and merged those changes.